core/src/aggregate.cpp

```diff
@@ -0,0 +1,63 @@
+#include <knoting/aggregate.h>
+#include <knoting/engine.h>
+
+namespace knot {
+namespace components {
+
+Aggregate::Aggregate() : m_aggregates(nullptr), m_aggregate(nullptr), m_Physics(nullptr), m_Scene(nullptr) {}
+Aggregate::~Aggregate() {}
+void Aggregate::on_awake() {
+    get_modules_from_physics();
+}
+void Aggregate::on_destory(){};
+
+void Aggregate::add_aggregate(std::string name, PxU32 number_actors, bool self_collision) {
+    for (int i = 0; i < m_aggregates->size(); i++) {
+        if (m_aggregates->at(i)->get_name() == name) {
+            log::error("already have same name aggregate");
+            return;
+        }
+    }
```

Comment:
https://en.cppreference.com/w/cpp/algorithm/find
https://stackoverflow.com/questions/571394/how-to-find-out-if-an-item-is-present-in-a-stdvector